lib/elixir/src/elixir_erl_pass.erl

```diff
@@ -400,6 +428,11 @@ translate_map(Meta, [{'|', _Meta, [Update, Assocs]}], S) ->
 translate_map(Meta, Assocs, S) ->
   translate_map(Meta, Assocs, none, S).
 
+translate_struct_var_name(Meta, Name, Args, S0) ->
```

Comment:
Instead of doing a pass later, we should change the struct generation to already add the guard.